pkg/executor/reaper/reaper.go

```diff
@@ -192,137 +227,159 @@ func CleanupRoleBindings(ctx context.Context, logger *zap.Logger, client kuberne
+						// if there's not even one function in the role-binding's namespace and there's not even
+						// one function with env reference to the SA's namespace, then remove that SA
+						// from this role-binding
+						if !ndmFunc && !funcEnvReference {
 							saToRemove[utils.MakeSAMapKey(subj.Name, subj.Namespace)] = true
 						}
 					}
-				} else if roleBinding.Name == fv1.SecretConfigMapGetterRB {
-					// if there's not even one function in the role-binding's namespace and there's not even
-					// one function with env reference to the SA's namespace, then remove that SA
-					// from this role-binding
-					if !ndmFunc && !funcEnvReference {
-						saToRemove[utils.MakeSAMapKey(subj.Name, subj.Namespace)] = true
-					}
 				}
-			}
-
-			// finally, make a call to RemoveSAFromRoleBindingWithRetries for all the service accounts that need to be removed
-			// for the role-binding in this iteration
-			if len(saToRemove) != 0 {
-				logger.Debug("removing service accounts from role binding",
-					zap.Any("service_accounts", saToRemove),
-					zap.String("role_binding_name", roleBinding.Name),
-					zap.String("role_binding_namespace", roleBinding.Namespace))
 
-				// call this once in the end for each role-binding
-				err = utils.RemoveSAFromRoleBindingWithRetries(ctx, logger, client, roleBinding.Name, roleBinding.Namespace, saToRemove)
-				if err != nil {
-					// if there's an error, we just log it and proceed with the next role-binding, hoping that this role-binding
-					// will be processed in next iteration.
-					logger.Debug("error removing service account from role binding",
-						zap.Error(err),
+				// finally, make a call to RemoveSAFromRoleBindingWithRetries for all the service accounts that need to be removed
+				// for the role-binding in this iteration
+				if len(saToRemove) != 0 {
+					logger.Debug("removing service accounts from role binding",
 						zap.Any("service_accounts", saToRemove),
 						zap.String("role_binding_name", roleBinding.Name),
 						zap.String("role_binding_namespace", roleBinding.Namespace))
+
+					// call this once in the end for each role-binding
+					err = utils.RemoveSAFromRoleBindingWithRetries(ctx, logger, client, roleBinding.Name, roleBinding.Namespace, saToRemove)
+					if err != nil {
+						// if there's an error, we just log it and proceed with the next role-binding, hoping that this role-binding
+						// will be processed in next iteration.
+						logger.Debug("error removing service account from role binding",
+							zap.Error(err),
+							zap.Any("service_accounts", saToRemove),
+							zap.String("role_binding_name", roleBinding.Name),
+							zap.String("role_binding_namespace", roleBinding.Namespace))
+					}
 				}
 			}
+			return nil
 		}
+		for _, namespace := range GetReaperNamespace() {
+			//ignore error
+			cleanupRoleBindings(namespace) //nolint errcheck
+		}
+	}
+}
+
+func GetReaperNamespace() map[string]string {
```

Comment:
Can we avoid mutations to `DefaultNSResolver()` outside of the module?
Other go modules should utilize values in read-only mode.

Only the utils package should be able to modify it.

Please consider moving this logic under ns resolver only.

Reply:
done